core/Cargo.toml

```diff
@@ -16,31 +16,41 @@ async-trait = "0.1.57"
 clap = { version = "4.0.9", features = ["derive"], optional = true }
 hex = { version = "0.4.3", default-features = false }
 log = "0.4.17"
-parity-scale-codec = "3.2.2"
+parity-scale-codec = { version = "3.6.1", features = ["derive"] }
 serde = "1.0.136"
 serde_json = "1.0.85"
 zstd = { version = "0.11.2", default-features = false }
 
 # Substrate
-frame-remote-externalities = { version = "0.28.0" }
-frame-try-runtime = { version = "0.27.0", features = ["try-runtime"] }
+frame-remote-externalities = { git = "https://github.com/paritytech/substrate", branch = "master" }
+frame-try-runtime = { git = "https://github.com/paritytech/substrate", branch = "master", features=["try-runtime"] }
 
-sc-cli = { version = "0.29.0" }
-sc-executor = { version = "0.25.0" }
+sc-cli = { git = "https://github.com/paritytech/substrate", branch = "master" }
+sc-executor = { git = "https://github.com/paritytech/substrate", branch = "master" }
 
-sp-api = { version = "19.0.0" }
-sp-core = { version = "20.0.0" }
-sp-externalities = { version = "0.18.0" }
-sp-io = { version = "22.0.0" }
-sp-keystore = { version = "0.26.0" }
-sp-runtime = { version = "23.0.0" }
-sp-rpc = { version = "19.0.0" }
-sp-state-machine = { version = "0.27.0" }
-sp-weights = { version = "19.0.0" }
+sp-api = { git = "https://github.com/paritytech/substrate", branch = "master" }
+sp-version = { git = "https://github.com/paritytech/substrate", branch = "master" }
+sp-core = { git = "https://github.com/paritytech/substrate", branch = "master" }
+sp-externalities = { git = "https://github.com/paritytech/substrate", branch = "master" }
+sp-io = { git = "https://github.com/paritytech/substrate", branch = "master" }
+sp-keystore = { git = "https://github.com/paritytech/substrate", branch = "master" }
+sp-runtime = { git = "https://github.com/paritytech/substrate", branch = "master" }
+sp-rpc = { git = "https://github.com/paritytech/substrate", branch = "master" }
+sp-state-machine = { git = "https://github.com/paritytech/substrate", branch = "master" }
+sp-weights = { git = "https://github.com/paritytech/substrate", branch = "master" }
 
-substrate-rpc-client = "0.26.0"
+substrate-rpc-client = { git = "https://github.com/paritytech/substrate", branch = "master" }
+
+[dev-dependencies]
+sc-service = { git = "https://github.com/paritytech/substrate", branch = "master" }
```

Comment:
sorry for nitpicking, but I'd keep all substrate deps together

more general, I'd maintain dependency order as:
<crates.io dependencies>
<git dependencies (Substrate ones)>
<path (local) dependencies>